sycl/doc/extensions/experimental/sycl_ext_intel_fp_control.asciidoc

```diff
@@ -0,0 +1,171 @@
+
+== Specification
+
+=== Feature test macro
+
+This extension provides a feature-test macro as described in the core SYCL
+specification.  An implementation supporting this extension must predefine the
+macro `SYCL_EXT_INTEL_FP_CONTROL` to one of the values defined in the table
+below.  Applications can test for the existence of this macro to determine if
+the implementation supports this feature, or applications can test the macro's
+value to determine which of the extension's features the implementation
+supports.
+
+
+[%header,cols="1,5"]
+|===
+|Value
+|Description
+
+|1
+|The APIs of this experimental extension are not versioned, so the
+ feature-test macro always has this value.
+|===
+
+=== Properties
+
+|===
+|Property|Description
+
+|`fp_control`
+|The `fp_control` kernel property provides a way to specify the rounding mode, denorm mode and single precision floating point mode for a kernel.
+|===
+
+The values that the property can be set to are a subset of the following that can be OR'ed together:
+
+[%header,cols="1,5"]
+|===
+|Mode | Description
+| round_to_nearest | Round to nearest or even
+| round_upward | Round towards positive infinity
+| round_downward | Round towards negative infinity
+| round_toward_zero | Round towards zero
+| denorm_ftz | Denorm mode flush to zero
+| denorm_d_allow | Denorm mode double allow
+| denorm_f_allow | Denorm mode float allow
+| denorm_hf_allow | Denorm mode half allow
+| denorm_allow | Denorm mode double/float/half allow
+|===
+
+The compiler will not enforce using mutually exclusive values (for instance 2 rounding
+modes) and the results in this case may be undefined. At most one `fp_control` property may be associated with a kernel. If a kernel is not associated with a `fp_control` property, the manner in which the fp mode is selected is implementation-defined.
+
+The properties are defined as follows:
+```c++
+namespace sycl::ext::intel::experimental {
+
+
+enum class fp_mode : /* implementation defined */ {
+  round_to_nearest = /* implementation defined */,
+  round_upward = /* implementation defined */,
+  round_downward = /* implementation defined */,
+  round_toward_zero = /* implementation defined */,
+
+  denorm_ftz = /* implementation defined */,
+  denorm_d_allow = /* implementation defined */,
+  denorm_f_allow = /* implementation defined */,
+  denorm_hf_allow = /* implementation defined */,
+  denorm_allow = /* implementation defined */
+};
+
+constexpr fp_mode operator|(const fp_mode &a, const fp_mode &b);
+
+struct fp_control_key {
+  template <fp_mode option>
+  using value_t = ext::oneapi::experimental::property_value<
+      fp_control_key, std::integral_constant<fp_mode, option>>;
+};
+
+template <fp_mode option = fp_mode::round_to_nearest>
```

Comment:
What is the rationale for having a default value here?  It seems to me that it would be better with no default.

Reply:
I agree, default value is not necessary, removed.